src/lib/sw.js

```diff
@@ -152,129 +112,34 @@ workboxRouting.registerRoute(
-    response = await offlinePartialResponse();
-  }
-
-  // If we can't get a real response (or the offline response), go to the
-  // network proper. This includes for 404 pages, as we don't want to mask a
-  // redirect handlers.
-  if (!response.ok) {
-    throw new Error(
-      `unhandled status for normal route '${url.pathname}': ${response.status}`,
-    );
-  }
-  const partial = await response.json();
-
-  const meta = partial.offline ? '<meta name="offline" value="true" />' : '';
-  const title = `<title>${partial.title || 'web.dev'}</title>`;
-  const rssHref = partial.rss || '/feed.xml';
-  const rssTitle =
-    partial.rss !== '/feed.xml'
-      ? `${partial.title} on web.dev`
-      : 'web.dev feed';
-  const rss = `<link rel="alternate" href="${rssHref}" type="application/atom+xml" data-title="${rssTitle}" />`;
-
-  const layoutTemplate = await templateForPartial();
-  const output = layoutTemplate
-    .replace('<!-- %_HEAD_REPLACE_% -->', `${meta}\n${title}\n${rss}`)
-    .replace('%_CONTENT_REPLACE_%', partial.raw);
-  const headers = new Headers();
-  headers.append('Content-Type', 'text/html');
-  return new Response(output, {headers, status: response.status});
+const pagePathRe = new RegExp('^/[\\w-/]*(?:|\\.html)$');
+const pageStrategy = new workboxStrategies.NetworkFirst({
+  cacheName: 'webdev-html-cache-v1',
+  plugins: [contentExpirationPlugin],
 });
 
-/**
- * @return {!Promise<string>} partial template to use in hydration
- */
-async function templateForPartial() {
-  const cachedResponse = await matchPrecache('/sw-partial-layout.partial');
-  if (!cachedResponse) {
-    // This occurs in development when the partial template isn't precached.
-    try {
-      return await fetch('/sw-partial-layout.partial');
-    } catch (e) {
-      console.warn('could not go to network for partial in dev', e);
-    }
-    return `<!DOCTYPE html>
-<head>
-<!-- %_HEAD_REPLACE_% -->
-</head>
-<body>
-<h1>web.dev dev partial</h1>
-%_CONTENT_REPLACE_%
-</body>
-</html>`;
-  }
-  return cachedResponse.text();
-}
+const pageMatch = matchSameOriginRegExp(pagePathRe);
+workboxRouting.registerRoute(pageMatch, pageStrategy);
 
 /**
- * @return {!Promise<Response>} response for the offline page's partial
+ * Cache images at runtime that aren't included in the original manifest, suchas author profiles.
  */
-async function offlinePartialResponse() {
-  const cachedResponse = await matchPrecache('/offline/index.json');
-  if (!cachedResponse) {
-    // This occurs in development when the offline partial isn't precached.
-    return new Response(
-      JSON.stringify({offline: true, raw: '<h1>Dev offline</h1>', title: ''}),
-    );
-  }
-  return cachedResponse;
-}
-
-workboxRouting.setCatchHandler(async ({url, request}) => {
-  // If we failed to fetch a partial, use the offline partial.
-  if (partialMatch({url})) {
-    return offlinePartialResponse();
-  }
+workboxRouting.registerRoute(
+  new RegExp('/images/.*'),
```

Comment:
I think this will cache author images but it won't cache any of the images used in articles as those would live in /blog/foo/bar.jpg

Should we cache those as well?

Reply:
ah right they come from imgix.